From: "Liam R. Howlett" <Liam.Howlett@oracle.com>
Subject: maple_tree: update forking to separate maple state and node
Date: Mon, 6 Nov 2023 10:45:51 -0500

Just a small change to the forking code after rebasing the maple status
changes on top.

Link: https://lkml.kernel.org/r/20231106154551.615042-1-Liam.Howlett@oracle.com
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Cc: Peng Zhang <zhangpeng.00@bytedance.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/maple_tree.c                 |    2 +-
 tools/testing/radix-tree/maple.c |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/lib/maple_tree.c~maple_tree-separate-ma_state-node-from-status-fix-2
+++ a/lib/maple_tree.c
@@ -6693,7 +6693,7 @@ static inline void mas_dup_build(struct
 
 	node = mt_alloc_one(gfp);
 	if (!node) {
-		new_mas->node = MAS_NONE;
+		new_mas->status = ma_none;
 		mas_set_err(mas, -ENOMEM);
 		return;
 	}
--- a/tools/testing/radix-tree/maple.c~maple_tree-separate-ma_state-node-from-status-fix-2
+++ a/tools/testing/radix-tree/maple.c
@@ -35956,7 +35956,7 @@ static int __init compare_tree(struct ma
 
 	if (mas_is_ptr(&mas_a) || mas_is_ptr(&mas_b)) {
 		if (!(mas_is_ptr(&mas_a) && mas_is_ptr(&mas_b))) {
-			pr_err("One is MAS_ROOT and the other is not.\n");
+			pr_err("One is ma_root and the other is not.\n");
 			return -1;
 		}
 		return 0;
@@ -35965,7 +35965,7 @@ static int __init compare_tree(struct ma
 	while (!mas_is_none(&mas_a) || !mas_is_none(&mas_b)) {
 
 		if (mas_is_none(&mas_a) || mas_is_none(&mas_b)) {
-			pr_err("One is MAS_NONE and the other is not.\n");
+			pr_err("One is ma_none and the other is not.\n");
 			return -1;
 		}
 
_
